bindings/python/pmix.pxi

```diff
@@ -903,6 +903,57 @@ cdef void pmix_free_pdata(pmix_pdata_t *array, size_t sz):
         n += 1
     PyMem_Free(array)
 
+# Convert a dictionary of key-value pairs into an
+# array of pmix_query_t structs
+#
+# @array [INPUT]
+#        - malloc'd array of pmix_query_t structs
+#
+# @pyq [INPUT]
+#          - a list of dictionaries, where each
+#            dictionary has a list of keys, and a
+#            list of qualifers
+#
+cdef int pmix_load_queries(pmix_query_t *array, pyq:list):
+    n = 0
+    for q in pyq:
+        nstrings = len(q['keys'])
+        if 0 < nstrings:
+            array[n].keys = <char **> PyMem_Malloc(nstrings * sizeof(char*))
+            if not array[n].keys:
+                return PMIX_ERR_NOMEM
+        # load keys from dictionary
+        n = 0
+        for k in q['keys']:
+            #keystr = str(k).encode('ascii')
+            array[n].keys[n] = strdup(k)
+            n += 1
+        array[n].nqual = len(q['qualifiers'])
+        rc = pmix_alloc_info(&(array[n].qualifiers), &(array[n].nqual), q['qualifiers'])
```

Comment:
I'm confused here. Why compute array[n].nqual and then call alloc_info with the address of that value? It appears that the function assigns the value, so no point in computing it first.

BTW: this is the location of the problem. Reuse of 'n' means that array[n] is now pointing into some random memory location.